File: leetcode/0137-single-number-ii.java
class Solution {
    public int singleNumber(int[] nums) {
        int ones = 0;
        int twos = 0;

        for (int n : nums) {
            ones = (~twos) & (ones ^ n);
            twos = (~ones) & (twos ^ n);
        }

        return ones;
    }
}

File: 0137-single-number-ii/0137-single-number-ii.ts
function singleNumber(nums: number[]): number {
    let ones: number = 0;
    let twos: number = 0;

    for (const n of nums) {
        ones = (~twos) & (ones ^ n);
        twos = (~ones) & (twos ^ n);
    }

    return ones;
}